### PR TITLE
Update plugin to comply with Yazi 4.0 standards

### LINKED ISSUE #2: Plugin not working after upgrading to yazi 0.4.2
I recently upgraded to yazi v0.4.2 and this plugin started giving errors of deprecation of few functions and linked me to https://github.com/sxyazi/yazi/issues/1772 and https://github.com/sxyazi/yazi/pull/1966.

### PR DESCRIPTION
**Pull request by Pahasara**
Resolves #2

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,23 +4,14 @@ Plugin for `Yazi` to preview all supported files via `Bat`.
...
-Previewing .rs files
-
-![Rust via bat](./yazi-bat--rust.png)
 
 ## Installation
 
 ```sh
 $> mkdir -p ~/.config/yazi/plugins
-$> git clone https://github.com/mgumz/yazi-plugin-bat.git ~/.config/yazi/plugins/bat.yazi
+$> git clone https://github.com/Pahasara/yazi-plugin-bat.git ~/.config/yazi/plugins/bat.yazi
```
Comment (mgumz) [inline README.md]: why would i merge this change of the url?

Comment (Pahasara) [reply README.md]: Sorry about that! I unintentionally left the URL pointing to my fork. I'll fix it right away and update the PR.

File: README.md [hunk 2]
```diff
@@ -4,17 +4,8 @@ Plugin for `Yazi` to preview all supported files via `Bat`.
 
 ## Preview
 
-Showing .csv files
+![preview](https://github.com/user-attachments/assets/67fe6e52-95c6-4768-a378-732e9c11cb84)
```
Comment (mgumz) [inline README.md]: The preview image is hosted outside of the repo. Please put the preview image into the repo

File: init.lua [hunk 3]
```diff
@@ -1,53 +1,54 @@
...
-    local child = Command("bat")
-        :args({
-            "--style", "plain",
-            "--color", "always",
-            tostring(self.file.url),
-        })
-        :stdout(Command.PIPED)
-        :stderr(Command.PIPED)
-        :spawn()
+function M:peek(job)
```
Comment (mgumz) [inline init.lua:3]: The change of the indentation level makes it hard t find the real difference between the versions.

Is the main change the "self" to "job" transition?

Comment (Pahasara) [reply init.lua:3]: Yeah, the main changes are the transition from self to job and replacing ui.Paragraph with ui.Text. The indentation issue has been fixed.

**Comment by mgumz** [comment]
@Pahasara Thank you very much!

**Pull request merged**